yarn-project/noir-libs/value-note/src/utils.nr

```diff
@@ -1,184 +1,125 @@
-        encrypted_data = change_note.serialise();
-    };
+    let mut note = ValueNote::new(amount, recipient);
+    create_note(context, balance, recipient, &mut note);
 
-    let encryption_pub_key = get_public_key(owner);
-    emit_encrypted_log(
-        context,
-        (*context).this_address(),
-        balance.storage_slot,
-        encryption_pub_key,
-        encrypted_data,
-    );
+    // It won't compile if Set.insert() is in an if statement :(
+    // if amount as u120 > 0 {
+    //     create_note(context, balance, recipient, &mut note);
+    // }
 }
 
-/*
-  Spends one note from a set of 4 notes.
-  Details: Reads 4 notes from the user's `balance`: [n_{o}, n_{o+1}, n_{o+2}, n_{o+3}]
-  where "o" is the note_offset. Then, spends the note: [n_{o + i}] 
-  where "i" is the spend_note_index. This gives more control to the user on which
-  note is to be spent. 
-*/
-fn spend_one_note(
+// Find some notes that add up to the amount.
+// Removes those note from owner's balance.
+// Fail if the sum of the selected notes is less than the amount.
+fn decrease_balance(
     context: &mut PrivateContext,
     balance: Set<ValueNote, VALUE_NOTE_LEN>,
     amount: Field,
     owner: Field,
-    spend_note_offset: u32,
 ) {
-    let options = NoteGetterOptions::new().set_limit(1).set_offset(spend_note_offset);
-    let opt_notes = balance.get_notes(context, options);
-
-    // The note should always exist.
-    let note = opt_notes[0].unwrap();
-
-    // Ensure the notes are actually owned by the owner (to prevent user from generating a valid proof while
-    // spending someone else's notes).
-    assert(owner == note.owner);
-
-    // Removes the note from the owner's set of notes.
-    balance.remove(context, note);
-
-    let note_value = note.value;
-    
-    // Assert that the note chosen to spend has enough funds.
-    assert(note_value as u64 >= amount as u64);
-
-    // Creates change note for the owner.
-    let change_value = note_value - amount;
-    let mut change_note = ValueNote::new(change_value, owner);
-
-    // Insert the change note to the owner's sets of notes.
-    balance.insert(context, &mut change_note);
-
-    // Emit the newly created encrypted note preimages via oracle calls.
-    let mut encrypted_data = [0; VALUE_NOTE_LEN];
-    if change_value != 0 {
-        encrypted_data = change_note.serialise();
-    };
-
-    let encryption_pub_key = get_public_key(owner);
-    emit_encrypted_log(
-        context,
-        context.inputs.call_context.storage_contract_address,
-        balance.storage_slot,
-        encryption_pub_key,
-        encrypted_data,
-    );
+    let sum = decrease_balance_max_amount(context, balance, amount, owner);
+    assert(sum == amount);
 }
 
-fn send_note(
+// Similar to `decrease_balance`, except that it doesn't fail if the subtracted amount is less than max_amount.
+// It returns the subtracted value, which should be less than or equal to max_amount.
+fn decrease_balance_max_amount(
```

Comment:
The name of this function confused me.
Taking into account my earlier comment about avoiding the word `balance` inside the `value_note/` dir...
perhaps it should be called something like `decrement_by_at_most` or `partially_decrement`?